.travis.yml

```diff
@@ -33,8 +33,8 @@ install:
 before_script:
   - set -o errexit; source .travis/test_05_before_script.sh
 script:
-  - if [ $SECONDS -gt 1200 ]; then set +o errexit; echo "Travis early exit to cache current state"; false; else set -o errexit; source .travis/test_06_script_a.sh; fi
-  - if [ $SECONDS -gt 1800 ]; then set +o errexit; echo "Travis early exit to cache current state"; false; else set -o errexit; source .travis/test_06_script_b.sh; fi
+  - if [ $SECONDS -gt 1200 ]; then set +o errexit; echo "Early exit to cache current state. Please re-run job"; false; else set -o errexit; source .travis/test_06_script_a.sh; fi
```

Comment:
I don't understand what "early exit to cache state means." Is this a timeout trying to create a cache, or a timeout waiting for something else to create a cache? What cache is being referring to? Where is the cache stored? Where can I find the code that creates the cache? Where I can I find the code or config that determines whether the cache needs to be updated and recreates the cache?

Reply:
@ryanofsky Travis stores a `ccache` to speed up the builds. If the build takes too long Travis will kill the job and I don't think it keeps the cache around when it does that. So, in order to keep the cache for future builds, we kill the job ourselves. Then on a rerun we should have that cache so it runs faster.

You can see the cache config for Travis here:
https://github.com/bitcoin/bitcoin/blob/master/.travis.yml#L4-L9

If anyone of that is incorrect someone please tell me, but that's how I understand it.

Reply:
Correct, aside from ccache, we also cache depends